src/app/server/Server.cpp

```diff
@@ -21,6 +21,11 @@
 
 #include <app/EventManagement.h>
 #include <app/InteractionModelEngine.h>
+#include <system/SystemConfig.h>
+#if defined CHIP_CONFIG_TC_REQUIRED_ACKNOWLEDGEMENTS && defined CHIP_CONFIG_TC_REQUIRED_ACKNOWLEDGEMENTS_VERSION
```

Comment:
CHIP_CONFIG variables are supposed to always be defined and just have values that mean something.  The checks for them should use `#if` on the value, and then compilers will catch the right config headers not being included, given our compiler flags.

This applies to all the places these config variables are used in this PR.